kore/src/Kore/Step/Remainder.hs

```diff
@@ -153,3 +167,50 @@ substitutionConditions subst =
   where
     substitutionCoverageWorker (x, t) =
         Syntax.Predicate.makeEqualsPredicate (mkVar x) t
+
+introduceDefinedness
+    :: forall variable
+    .  ( SortedVariable variable
+       , FreshVariable variable
+       , Unparse variable
+       , Show variable
+       )
+    => OrPredicate variable
+    -> Pattern variable
+    -> Pattern variable
+introduceDefinedness cond result =
+    foldr (flip andCondition) result cond
+
+ceilChildOfApplication
+    :: forall variable simplifier
+    .  ( FreshVariable variable
+       , SortedVariable variable
+       , Show variable
+       , Unparse variable
+       , MonadSimplify simplifier
+       )
+    => TermLike variable
+    -> simplifier (OrPredicate variable)
+ceilChildOfApplication =
+    withCeilChildOfApplication
```

Comment:
Why is having the `with...` function better than using its body here?

Reply:
I used to need that function for some failing tests, but after making the changes suggested by you last week the tests started working again.